pkg/credentialprovider/aws/aws_credentials.go

```diff
@@ -41,7 +41,7 @@ import (
 )
 
 var (
-	ecrPattern = regexp.MustCompile(`^(\d{12})\.dkr\.ecr(\-fips)?\.([a-zA-Z0-9][a-zA-Z0-9-_]*)\.amazonaws\.com(\.cn)?$`)
+	ecrPattern = regexp.MustCompile(`^(\d{12})\.dkr\.ecr(\-fips)?\.([a-zA-Z0-9][a-zA-Z0-9-_]*)\.(amazonaws\.com(\.cn)?|sc2s\.sgov\.gov|c2s\.ic\.gov)$`)
```

Comment:
@andrewsykim can you confirm the credential provider plugin can handle logic like this?

Reply:
@liggitt we've had this patch in prod for quite a bit of time from the looks of it.

- https://github.com/aws/aws-sdk-js/blame/master/lib/region_config_data.json#L27-L36
- https://github.com/hashicorp/terraform-provider-aws/issues/19000
- https://github.com/hashicorp/terraform-provider-aws/issues/19014
- https://github.com/hashicorp/terraform-provider-aws/issues/9993

Funnily we have it in our documentation too :)  https://kubernetes.io/docs/tasks/kubelet-credential-provider/kubelet-credential-provider/

Reply:
> Funnily we have it in our documentation too :) https://kubernetes.io/docs/tasks/kubelet-credential-provider/kubelet-credential-provider/

ha (though that example is more constrained than the entire subdomain .. only ".dkr.ecr.us-iso-east-1.c2s.ic.gov" and ".dkr.ecr.us-isob-east-1.sc2s.sgov.gov" suffixes, which are much more clearly ecr

Reply:
> Funnily we have it in our documentation too :) https://kubernetes.io/docs/tasks/kubelet-credential-provider/kubelet-credential-provider/

This is documentation for the external credential provider though, which is to demonstrate that you can configure match images with whatever images you want and removes the need for us to continuously update the internal plugin with new subdomains.